homeassistant/components/owlet/config_flow.py

```diff
@@ -0,0 +1,188 @@
+        vol.Required("username"): str,
+        vol.Required("password"): str,
+    }
+)
+
+
+class OwletConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Owlet Smart Sock."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialise config flow."""
+        self._entry: ConfigEntry
+        self._region: str
+        self._username: str
+        self._password: str
+        self._devices: dict[str, Sock]
+        self.reauth_entry: ConfigEntry | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            self._region = user_input[CONF_REGION]
+            self._username = user_input[CONF_USERNAME]
+            self._password = user_input[CONF_PASSWORD]
+
+            owlet_api = OwletAPI(
+                region=self._region,
+                user=self._username,
+                password=self._password,
+                session=async_get_clientsession(self.hass),
+            )
+
+            await self.async_set_unique_id(self._username.lower())
+            self._abort_if_unique_id_configured()
+
+            try:
+                token = await owlet_api.authenticate()
+                try:
+                    await owlet_api.validate_authentication()
+                    return self.async_create_entry(
+                        title=self._username,
+                        data={
+                            CONF_REGION: self._region,
+                            CONF_USERNAME: self._username,
+                            CONF_API_TOKEN: token[CONF_API_TOKEN],
+                            CONF_OWLET_EXPIRY: token[CONF_OWLET_EXPIRY],
+                            CONF_OWLET_REFRESH: token[CONF_OWLET_REFRESH],
+                        },
+                        options={CONF_SCAN_INTERVAL: POLLING_INTERVAL},
+                    )
+                except OwletDevicesError:
+                    errors["base"] = "no_devices"
+
+            except OwletEmailError:
+                errors["base"] = "invalid_email"
+            except OwletPasswordError:
+                errors["base"] = "invalid_password"
+            except OwletCredentialsError:
+                errors["base"] = "invalid_credentials"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+
+        return self.async_show_form(
+            step_id="user", data_schema=STEP_USER_DATA_SCHEMA, errors=errors
+        )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry: ConfigEntry) -> OptionsFlowHandler:
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+    async def async_step_reauth(self, user_input: Mapping[str, Any]) -> FlowResult:
```

Comment:
You don't have to, but I would probably remove reauth for now and it add it in a future release. It is a helpful thing to have but it is not the top priority.

And again the smaller the pr the more likely it is for it to get merged faster.